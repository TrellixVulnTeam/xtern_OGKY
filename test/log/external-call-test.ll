; TODO: fix pid in log name
; XFAIL: *
; RUN: %projbindir/tern-instr < %s -S -o %t1
; RUN: llc -o %t1.s %t1-record.ll
; RUN: g++ -o %t1 %t1.s %ternruntime -lpthread -lrt
; RUN: ./%t1
; RUN: rm -rf %t1.outdir
; RUN: env TERN_OPTIONS=log_type=bin:output_dir=%t1.outdir ./%t1
; RUN: %projbindir/logprint -bc %t1-analysis.ll %t1.outdir/tid-0.bin -r -v | FileCheck %s.out

; ModuleID = 'two-transfers-test.ll'
target datalayout = "e-p:64:64:64-i1:8:8-i8:8:8-i16:16:16-i32:32:32-i64:64:64-f32:32:32-f64:64:64-v64:64:64-v128:128:128-a0:0:64-s0:64:64-f80:128:128-n8:16:32:64"
target triple = "x86_64-unknown-linux-gnu"

%struct.timeval = type { i64, i64 }

define i32 @main() nounwind {
entry:
  %FullTime = alloca %struct.timeval, align 8
  %0 = call i32 @gettimeofday(%struct.timeval* noalias %FullTime, i8* noalias null) nounwind
  %1 = getelementptr inbounds %struct.timeval* %FullTime, i64 0, i32 1
  %2 = load i64* %1, align 8
  br label %return

return:
  ret i32 %0
}

declare i32 @gettimeofday(%struct.timeval* noalias, i8* noalias) nounwind
