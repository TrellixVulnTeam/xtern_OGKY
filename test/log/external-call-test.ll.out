CHECK: ins=n/a sync tern_thread_begin() turn = 0
CHECK: ins=1 call gettimeofday([[ADDRINDEC:[1-9][0-9]+]], 0, 0 more args)
CHECK: ins=1 ret gettimeofday() = 0
CHECK: ins=3 load [[ADDR:0x[0-9a-f]+]] = [[VALUE:[1-9][0-9]+]]
CHECK: ins=5 marker
CHECK: ins=n/a sync tern_thread_end() turn = 1
CHECK: P ins=0  %FullTime = alloca %struct.timeval, align 8, !ins_id !0 ; <%struct.timeval*> [#uses=2]
CHECK: L idx=1 ins=1 call gettimeofday([[ADDRINDEC]], 0) = 0  %0 = call i32 @gettimeofday(%struct.timeval* noalias %FullTime, i8* noalias null) nounwind, !ins_id !1 ; <i32> [#uses=1]
CHECK: P ins=2  %1 = getelementptr inbounds %struct.timeval* %FullTime, i64 0, i32 1, !ins_id !2 ; <i64*> [#uses=1]
CHECK: L idx=3 ins=3 load [[ADDR]] = [[VALUE]] %2 = load i64* %1, align 8, !ins_id !3          ; <i64> [#uses=0]
CHECK: P ins=4  br label %return, !ins_id !4
CHECK: P ins=5  ret i32 %0, !ins_id !5
